common/os_calls.c

```diff
@@ -416,11 +416,27 @@ g_tcp_socket(void)
 
 #if defined(XRDP_ENABLE_IPV6)
     rv = (int)socket(AF_INET6, SOCK_STREAM, 0);
+    if (rv < 0)
+    {
+        log_message(LOG_LEVEL_ERROR, "g_tcp_socket: %s", strerror(errno));
+
+        switch(errno)
+        {
+            case EAFNOSUPPORT: /* if IPv6 not supported, retry IPv4 */
+                log_message(LOG_LEVEL_INFO, "IPv6 not supported, falling back to IPv4");
```

Comment:
I mean the caller of the socket may have an idea whether it wants an IPv4 socket or and IPv6 socket, based on what it wants to do to the socket (which may depend on the configuration). But I looked closer at the code. It looks like os_calls.c tries to make IP protocol selection transparent to the caller for all calls. It remains to be seen how well it would work, but it's pre-existing code.

I have no objections against your patch. Just please add space after `switch` and after `(int)`